account_multicurrency_revaluation/report/currency_unrealized_report.py

```diff
@@ -0,0 +1,94 @@
+                            'gl_ytd_balance']
+
+    def __contains__(self, key):
+        return hasattr(self, key)
+
+    def get_lines(self):
+        """Get all line account move line that are need on report for current
+        account.
+        """
+        sql = """SELECT res_partner.name,
+                   account_move_line.date,
+                   account_move_line.gl_foreign_balance,
+                   account_move_line.gl_currency_rate,
+                   account_move_line.gl_revaluated_balance,
+                   account_move_line.gl_balance,
+                   account_move_line.gl_revaluated_balance -
+                   account_move_line.gl_balance as gl_ytd_balance,
+                   res_currency.name as curr_name
+                 FROM account_move_line
+                   LEFT join res_partner on
+                     (account_move_line.partner_id = res_partner.id)
+                   LEFT join account_move on
+                     (account_move_line.move_id = account_move.id)
+                   LEFT join res_currency on
+                     (account_move_line.currency_id = res_currency.id)
+                 WHERE account_move_line.account_id = %s
+                   AND account_move.to_be_reversed = true
+                   AND account_move_line.gl_balance is not null
+                 ORDER BY res_partner.name,
+                   account_move_line.gl_foreign_balance,
+                   account_move_line.date"""
+        self.cursor.execute(sql, [self.account_id])
+        self.ordered_lines = self.cursor.dictfetchall()
+        return self.ordered_lines
+
+    def compute_totals(self):
+        """Compute the sum of values in self.ordered_lines"""
+        totals = dict.fromkeys(self.keys_to_sum, 0.0)
+        for line in self.ordered_lines:
+            for tot in self.keys_to_sum:
+                totals[tot] += line.get(tot, 0.0)
+        for key, val in totals.items():
+            setattr(self, key + '_total', val)
+
+
+class CurrencyUnrealizedReport(models.AbstractModel):
+    _name = 'report.account_multicurrency_revaluation_report.curr_unrealized'
+
+    @api.model
+    def render_html(self, docids, data=None):
+        shell_accounts = {}
+        docs = self.env['account.account']
+        data = data if data is not None else {}
+        accounts = self.env['account.account'].search(
+            [('currency_revaluation', '=', True)])
+        for account in accounts:
+            acc = ShellAccount(account)
+            acc.get_lines()
+            if acc.ordered_lines:
+                docs |= account
+                shell_accounts[account.id] = acc
+                acc.compute_totals()
+        docargs = {
+            'doc_ids': docs.ids,
+            'doc_model': 'account.account',
+            'docs': docs,
+            'shell_accounts': shell_accounts,
+            'data': dict(
+                data,
+            ),
+        }
+        return self.env['report'].render(self._name[7:], docargs)
```

Comment:
This won't work in Odoo 11.0

What is needed is to use the new report structure. `report` module has been splitted between `web` and `base` modules.
The good new is that it is now pretty simple to render a specific report. You need to use the xmlid:

``` python
self.env.ref('foo_report_xmild').render(recordset, data=data)
```

Thus in your case, try:

``` python
self.env.ref('account_multicurrency_revaluation.curr_unrealized').render(docs)
```